Issue/pr create: exit with nonzero status code when "Cancel" was chosen

mislav opened the pull request:
This is to indicate that the command had not finished successfully.

Ref. https://github.com/cli/cli/issues/3002

mislav committed:
issue/pr create: exit with nonzero status code when "Cancel" was chosen

This is to indicate that the command had not finished successfully.

mislav committed:
Avoid triggering recovery mechanism when cancelling `issue/pr create`

mislav committed:
Avoid the issue/pr recovery mechanism handling Ctrl-C keypress in prompts

Either InterruptErr or SilentErr will be present when the user has
chosen "Cancel" or pressed Ctrl-C in prompts. We don't want the recovery
mechanism to kick in these cases because the cancellation was likely
willingly initiated by the user.

mislav commented:
@vilmibm I've pushed some tweaks regarding avoiding the PreserveInput mechanism in these cases. Please take a look!

vilmibm reviewed (approved):
ah, good catch.

I'm a little nervous about `SilentError` meaning user cancellation implicitly but am fine with it for now :+1:

mislav committed:
Exit with status code "2" on user cancellation errors

This also stops printing "interrupt" after Ctrl-C is pressed.

mislav committed:
Merge remote-tracking branch 'origin' into cancel-error-status

mislav commented:
> I'm a little nervous about `SilentError` meaning user cancellation implicitly but am fine with it for now 👍

That's a great point. I've now introduced a new concept of "CancelError" which, together with "interrupt" error when <kbd>Ctrl-C</kbd> is pressed during Survey, always means explicit user cancellation. For scripts to be able to detect the operation being explicitly cancelled by the user, the `gh` process now exits with status code "2" in those cases instead of status code "1".

vilmibm reviewed (approved):
oh awesome, thank you! I feel much better about this.

mislav committed:
Formalize `gh` process exit codes

Here are the statuses:
- 0: success
- 1: misc. error
- 2: user interrupt/cancellation
- 4: authentication needed

These old exit codes are now changed to "1":
- we used to return "2" for config file errors;
- we used to return "2" for alias expansion errors;
- we used to return "3" for alias runtime errors.

I do not believe that there is a need to distinguish these specific
cases via exit status, and converting them to "1" frees codes "2" and
"3" for more practical use.
